File: CornerGrocerCode.py
# Choyce Reynolds
# 10/19/2025

# I wanted Python to hadnle all of the logic for this program. It pulls the information from the input file, C++ calls the information based on user input, and the Python writes the output file

import sys

class ItemFrequency:
    def __init__(self):
        self.frequency = {}  
        with open("InputFile.txt", "r") as file:
            items = file.read().splitlines()
        for item in items:
            if item in self.frequency:
                self.frequency[item] += 1
            else:
                self.frequency[item] = 1

    # Shows all itmes and counts their quantities
    def CountAllItems(self):
        for item, count in self.frequency.items():
            print(f"{item}: {count}")

    # Shows one item and counts its quantity
    def CountOneItem(self, ItemName):
        count = self.frequency.get(ItemName, 0)
        print(f"{ItemName}: {count}")

     # Puts items and their quantities into a data file
    def WriteFrequencyFile(self, OutputFile = "frequency.dat"):
        with open("frequency.dat", "w") as out:
            for item, count in self.frequency.items():
                out.write(f"{item} {count}\n")


    # Shows a histogram of items and their quantities    
    def ShowHistogram(self):
        self.WriteFrequencyFile()
        with open("frequency.dat", "r") as file:
            for line in file:
                item, count = line.strip().split()
                print(f"{item:15} {'*' * int(count)}")

# Handles the arguments that are called in the cpp file
if __name__ == "__main__":
    FrequencyMap = ItemFrequency()

    if len(sys.argv) < 2:
        print("Usage: python CornerGrocerCode.py [all|one <item>|histogram]")
    elif sys.argv[1] == "all":
        FrequencyMap.CountAllItems()
    elif sys.argv[1] == "one" and len(sys.argv) == 3:
        FrequencyMap.CountOneItem(sys.argv[2])
    elif sys.argv[1] == "histogram":
        FrequencyMap.ShowHistogram() 

File: CornerGrocerCode.cpp
// Choyce Reynolds
// 10/19/2025

// I wanted C++ to handle displaying the menue and calling logic from python in order to display the information based on user input

#include <iostream>
#include <string>
#include <cstdlib> 

using namespace std;

// Displays the menu where users make their selections
void displayMenu() {
    cout << "***********************************" << endl;
    cout << "||     Corner Grocer Items       ||" << endl;
    cout << "***********************************" << endl;
    cout << "Please make a selection from the following options:" << endl;
    cout << "1. Count all items" << endl;
    cout << "2. Count one item" << endl;
    cout << "3. Show histogram" << endl;
    cout << "4. Exit menu" << endl;
    cout << "Enter your selection here: ";
}

int main() {
    int input;
    string item;

    // This loop will continue until the user exits the menu by entering 4
    do {
        displayMenu();
        cin >> input;

        switch (input) {
        case 1:
            system("python CornerGrocerCode.py all"); // Calls all items with their quantities from the python file
            break;
        case 2:
            cout << "Enter item name: ";
            cin >> item;
            system(("python CornerGrocerCode.py one " + item).c_str()); // Calls one item and its quantity from the python file
            break;
        case 3:
            system("python CornerGrocerCode.py histogram"); // Calls the histogram from the python file
            break;
        case 4:
            cout << "Exiting menu" << endl; // Exits the menu
            break;
        default:
            cout << "I'm sorry, but that is an invalid choice. Please try again." << endl; // This handles incorrect input from the user
        }
    } while (input != 4);

    return 0;
}
